### DISCUSSION TITLE
[Integration] From DSLs to Accelerator-Rich Platform Implementations: Addressing the Mapping Gap

### DISCUSSION THREAD

**Discussion by slyubomirsky** [cucapra/latte21#27] [Papers]
Authors: Bo-Yuan Huang, Steven Lyubomirsky, Thierry Tambe, Yi Li, Mike He, Gus Smith, Gu-Yeon Wei, Aarti Gupta, Sharad Malik, Zachary Tatlock
Paper PDF: https://capra.cs.cornell.edu/latte21/paper/30.pdf
Talk Recording: https://www.youtube.com/watch?v=2U7RAeOzssA

**Comment by stephenneuendorffer** [comment]
I think it would be very interesting to see if the conversion from ML models to ILA could be made more structured using MLIR end-to-end rather than the TVM BYOC backend.

**Reply by slyubomirsky** [reply]
Yes, that's definitely something we've thought about as well. TVM and BYOC presented many conveniences in terms of inheriting deep learning infrastructure, but as long as there is a step that outputs ILA instructions, any other compiler framework could be used to implement our proposal

**Comment by ekiwi** [comment]
Since I did not see it mentioned anywhere in the paper: Are you planning to do a refinement proof between your ILA and one of the hardware accelerator implementations?

**Reply by slyubomirsky** [reply]
My apologies for not seeing this comment at the time, but that is something we are actively working on.
